app/controllers/api/account.php

```diff
@@ -535,13 +593,59 @@
                         'search' => implode(' ', [$userId, $email, $name])
                     ])));
                 } catch (Duplicate $th) {
-                    throw new Exception(Exception::USER_ALREADY_EXISTS);
+                    $failureRedirect(Exception::USER_ALREADY_EXISTS);
                 }
             }
         }
 
+        Authorization::setRole(Role::user($user->getId())->toString());
+        Authorization::setRole(Role::users()->toString());
+
         if (false === $user->getAttribute('status')) { // Account is blocked
-            throw new Exception(Exception::USER_BLOCKED); // User is in status blocked
+            $failureRedirect(Exception::USER_BLOCKED); // User is in status blocked
+        }
+
+        $identity = $dbForProject->findOne('identities', [
```

Comment:
To the existing userInternalId index , you can add these 2 attributes: provider, providerUid
['userInternalId', 'provider', 'providerUid']

Reply:
You're suggesting to make sure there's an index on this query (userInternalId, provider, and providerUid)?

Reply:
yes , since there is a unique index on [provider,providerUid],
I think adding an additional index on the above will be good